Add samsung-klte mainline kernel

drebrez opened the pull request:
Changes:
- `device-samsung-klte` with both downstream and mainline kernel subpackages
- `linux-postmarketos-qcom` updated to last commit

What works in mainline kernel:
- internal SD card
- volume and home key buttons
- usb network

---
[x] Merge on GitHub (see <https://postmarketos.org/merge>)

drebrez commented:
Just tested and I can confirm that it doesn't work.
Currently the boot.img is created with QCDT (dtb inside the boot.img) and also the kernel with the appended dtb.
I tried with QCDT and the kernel without the appended dtb but it doesn't work either.
In just shows the samsung download mode with an error message.

With `deviceinfo_bootimg_qcdt="false"` and the dtb appended to the kernel it boots fine.

FYI: the current version of linux-postmarketos-qcom doesn't have a usb node for the samsung-klte, I will add it very soon.

ollieparanoid commented:
Thanks for the PR!
* Regarding split of the deviceinfo package, see my answer in https://github.com/postmarketOS/pmbootstrap/issues/1368#issuecomment-387881831
* Maybe it makes sense to ship the deviceinfo package split as its own PR, so we could merge it even if running mainline that way does not work with your device yet?

drebrez commented:
I thought about splitting the PR but it was useful to have the other files
to test it. I will split the PR later.
Tomorrow I will look at the your version

On Wed, 9 May 2018, 23:34 Oliver Smith, <notifications@github.com> wrote:

> Thanks for the PR!
>
>    - Regarding split of the deviceinfo package, see my answer in #1368
>    (comment)
>    <https://github.com/postmarketOS/pmbootstrap/issues/1368#issuecomment-387881831>
>    - Maybe it makes sense to ship the deviceinfo package split as its own
>    PR, so we could merge it even if running mainline that way does not work
>    with your device yet?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/postmarketOS/pmbootstrap/pull/1479#issuecomment-387882703>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABiXGxtAuMOJ-HsSnzg937FAISStnbw6ks5tw2DvgaJpZM4T42Wf>
> .
>

opendata26 commented:
As for qcdt, decompile a downstream DTS and add the properties at the top,(msm-id and a few others) to your upstream dts,

opendata26 commented:
@drebrez yeah

drebrez commented:
@ollieparanoid I've rebased this PR to only include the mainline kernel for the samsung-klte, It works with the same `deviceinfo` as the downstream kernel since I've enabled the QCDT support and also the usb network.
I will create a new PR for the `devicepkg-dev` with the same version I've added here before and we can continue the discussion there.
